lib/ldap.c

```diff
@@ -183,38 +183,43 @@ const struct Curl_handler Curl_handler_ldaps = {
 #endif
 
 #ifdef USE_WIN32_LDAP
-static int Win_ldap_bind(struct connectdata *conn, LDAP *server, TCHAR *user, TCHAR *passwd)
+static int Win_ldap_bind(struct connectdata *conn,
+                         LDAP *server,
+                         TCHAR *user,
+                         TCHAR *passwd)
 {
-    int rc = 0;
-    ULONG method = LDAP_AUTH_NEGOTIATE;
-    SEC_WINNT_AUTH_IDENTITY cred = { 0 };
-    PCHAR inuser = NULL;
-    PCHAR inpass = NULL;
-
-    if (user != NULL && passwd != NULL) {
-        cred.User = user;
-        cred.UserLength = strlen(user);
-        cred.Password = passwd;
-        cred.PasswordLength = strlen(passwd);
-        cred.Flags = SEC_WINNT_AUTH_IDENTITY_ANSI;
-        if (conn->data->set.httpauth & CURLAUTH_BASIC) {
-            method = LDAP_AUTH_SIMPLE;
-            inuser = user;
-            inpass = passwd;
-        }
-        else if (conn->data->set.httpauth & CURLAUTH_DIGEST) {
-            method = LDAP_AUTH_DIGEST;
-            inpass = (TCHAR*)(&cred);
-        }
-        else if (conn->data->set.httpauth & CURLAUTH_NTLM) {
-            method = LDAP_AUTH_NTLM;
-            inpass = (TCHAR*)(&cred);
-        }
-        else {
-            inpass = (TCHAR*)(&cred);
-        }
+  ULONG method = LDAP_AUTH_NEGOTIATE;
+  SEC_WINNT_AUTH_IDENTITY cred = { 0 };
+  PCHAR inuser = NULL;
+  PCHAR inpass = NULL;
+
+  if(user != NULL && passwd != NULL) {
+    cred.User = user;
+    cred.UserLength = _tcslen(user);
+    cred.Password = passwd;
+    cred.PasswordLength = _tcslen(passwd);
+    cred.Flags = (sizeof(TCHAR) == sizeof(char)
+                ? SEC_WINNT_AUTH_IDENTITY_ANSI
+                : SEC_WINNT_AUTH_IDENTITY_UNICODE);
```

Comment:
checksrc and [code style](https://github.com/curl/curl/blob/master/docs/CODE_STYLE.md) may not cover this but the operator goes on the end if it's over multiple lines, like
~~~
a &&
b

longfoo ? longbar :
longbaz;

longfoo ? longbar :
          longbaz; (rare)

longfoo ?
longbar :
longbaz;

in the repo there's also another way with a 2 space indent from the starting
column but i think that's some old style
var = x ?
  y : z;
~~~
so you could change it to
~~~
    cred.Flags = (sizeof(TCHAR) == sizeof(char)) ?
                 SEC_WINNT_AUTH_IDENTITY_ANSI :
                 SEC_WINNT_AUTH_IDENTITY_UNICODE;
~~~

the user != NULL it's preferred just `if(user && passwd)`, though != NULL appears to be accepted. I did a git grep just now and there are numerous instances.

also args can be squashed to two lines
~~~
static int Win_ldap_bind(struct connectdata *conn, LDAP *server,
                         TCHAR *user, TCHAR *passwd)
~~~

as i mentioned someone else will have to +1. i think it would be worthwhile to explain how this behavior is different from ldap in linux, if it is not obvious ( i don't use ldap)